src/_P042_Candle.ino

```diff
@@ -679,87 +718,93 @@ void type_ColorFader() {
 void HSVtoRGB(int hue, int sat, int val, int colors[3]) {
   // hue: 0-359, sat: 0-255, val (lightness): 0-255
-  int r=0, g=0, b=0, base=0;
+  int r = 0, g = 0, b = 0, base = 0;
 
   if (sat == 0) { // Achromatic color (gray).
-    colors[0]=val;
-    colors[1]=val;
-    colors[2]=val;
+    colors[0] = val;
+    colors[1] = val;
+    colors[2] = val;
   }
   else  {
-    base = ((255 - sat) * val)>>8;
-    switch(hue/60) {
-    case 0:
-      r = val;
-      g = (((val-base)*hue)/60)+base;
-      b = base;
-      break;
-    case 1:
-      r = (((val-base)*(60-(hue%60)))/60)+base;
-      g = val;
-      b = base;
-      break;
-    case 2:
-      r = base;
-      g = val;
-      b = (((val-base)*(hue%60))/60)+base;
-      break;
-    case 3:
-      r = base;
-      g = (((val-base)*(60-(hue%60)))/60)+base;
-      b = val;
-      break;
-    case 4:
-      r = (((val-base)*(hue%60))/60)+base;
-      g = base;
-      b = val;
-      break;
-    case 5:
-      r = val;
-      g = base;
-      b = (((val-base)*(60-(hue%60)))/60)+base;
-      break;
+    base = ((255 - sat) * val) >> 8;
+
+    switch (hue / 60) {
+      case 0:
+        r = val;
+        g = (((val - base) * hue) / 60) + base;
+        b = base;
+        break;
+      case 1:
+        r = (((val - base) * (60 - (hue % 60))) / 60) + base;
+        g = val;
+        b = base;
+        break;
+      case 2:
+        r = base;
+        g = val;
+        b = (((val - base) * (hue % 60)) / 60) + base;
+        break;
+      case 3:
+        r = base;
+        g = (((val - base) * (60 - (hue % 60))) / 60) + base;
+        b = val;
+        break;
+      case 4:
+        r = (((val - base) * (hue % 60)) / 60) + base;
+        g = base;
+        b = val;
+        break;
+      case 5:
+        r = val;
+        g = base;
+        b = (((val - base) * (60 - (hue % 60))) / 60) + base;
+        break;
     }
-    colors[0]=r;
-    colors[1]=g;
-    colors[2]=b;
+    colors[0] = r;
+    colors[1] = g;
+    colors[2] = b;
   }
 }
 
 // Convert RGB Color to HSV Color
 void RGBtoHSV(uint8_t r, uint8_t g, uint8_t b, double hsv[3]) {
-    double rd = (double) r/255;
-    double gd = (double) g/255;
-    double bd = (double) b/255;
-    double maxval = rd;
-    if (gd > maxval) { maxval = gd; }
-    if (bd > maxval) { maxval = bd; }
-    double minval = rd;
-    if (gd < minval) { minval = gd; }
-    if (bd < minval) { minval = bd; }
-    double h = 0, s, v = maxval;
-    double d = maxval - minval;
-
-    s = maxval == 0 ? 0 : d / maxval;
-
-    if (maxval == minval) {
-        h = 0; // achromatic
-    } else {
-        if (maxval == rd) {
-            h = (gd - bd) / d + (gd < bd ? 6 : 0);
-        } else if (maxval == gd) {
-            h = (bd - rd) / d + 2;
-        } else if (maxval == bd) {
-            h = (rd - gd) / d + 4;
-        }
-        h /= 6;
+  double rd     = (double)r / 255;
+  double gd     = (double)g / 255;
+  double bd     = (double)b / 255;
+  double maxval = rd;
```

Comment:
Is `double` resolution really needed here? If this function is called often, it might be faster working on floats.

Reply:
Thats the TODO part... 😃

Reply:
Replaced the `double`s by `float`s and moved as RGB2HSV to `Misc.cpp` for generic use.